### PR TITLE
Portfolio API & Tests

### LINKED ISSUE #243: Implement Portfolio API and Tests (Req 1.1.6.)
## Prerequisite

- [x] I am using the latest version of the code, and checked the documentation related to my needs.

* **I'm submitting a ... (remove the irrelevant one)**
  - [x] feature request
  - [x] test request

* **Briefly describe the nature of the issue. If test request, please provide the functionality. If feature request, please provide related requirement.**
Requirement 1.1.6.

### PR DESCRIPTION
**Pull request by harun-eren**
# Pull Request Template

## Description

Please include a summary of the work done and which issue is addressed.

Fixes #243

## Type of change

Please delete options that are not relevant.

- [x] New feature (new functionality)

## Checklist:

- [ ] My code follows pep8
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation, or created a related issue.
- [ ] I have committed tests that prove my fix is effective or that my feature works. Or, I have opened an issue.
- [ ] I have added 2 reviewers to get the approval of.

### CODE REVIEW

File: traiders/backend/api/models/portfolio.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from django.db import models
+from .users import User
+from .parity import Parity
+
+
+class Portfolio(models.Model):
+    name = models.CharField(max_length=64, blank=False)
+    owner = models.ForeignKey(User, on_delete=models.CASCADE, blank=False)
```
Comment (yemreinci) [inline traiders/backend/api/models/portfolio.py]: Can you rename `owner` to `user`? We followed the same convention in other models too, it is clear that the user "owns" the portfolio.

File: traiders/backend/api/views/portfolio.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        if 'user' not in request.query_params:
+            return Response({
+                'user': 'Please provide a user to see his/her portfolios'
+            }, status=status.HTTP_400_BAD_REQUEST)
+
+        return super().list(request, *args, **kwargs)
+
+    def check_object_permissions(self, request, portfolio):
+        # Another user can only retrieve; cannot update, delete, update or partial_update
+        if self.action != 'retrieve' and request.user != portfolio.owner:
```
Comment (yemreinci) [inline traiders/backend/api/views/portfolio.py]: Other users can update or partial_update the `is_following` field. Let other users partial_update and update as well.

File: traiders/backend/api/serializers/portfolio.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+class PortfolioSerializer(serializers.HyperlinkedModelSerializer):
+    is_following = IsFollowingField()
+
+    class Meta:
+        model = Portfolio
+        fields = ["name", "owner", "followed_by", "parities"]
+
+    def update(self, instance: Portfolio, validated_data: dict):
+        is_following = validated_data.pop('is_following', None)
```
Comment (yemreinci) [inline traiders/backend/api/serializers/portfolio.py]: TODO
1 - Check whether the owner of the portfolio is trying to follow their own portfolio. Raise permission error.
2 - Check whether some user other than the owner trying the update a field other than `is_following`. i.e: another user cant change the `name`. Raise permission error.

File: traiders/backend/api/views/portfolio.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+class PortfolioViewSet(mixins.CreateModelMixin,
+                       mixins.RetrieveModelMixin,
+                       mixins.UpdateModelMixin,
+                       mixins.ListModelMixin,
+                       mixins.DestroyModelMixin,
+                       GenericViewSet):
+    serializer_class = PortfolioSerializer
+    queryset = Portfolio.objects.all()
+    filter_backends = [DjangoFilterBackend]
+
```
Comment (yemreinci) [inline traiders/backend/api/views/portfolio.py]:
```suggestion
permission_classes = (IsAuthenticatedOrReadOnly,)
```

Comment (harun-eren) [reply traiders/backend/api/views/portfolio.py]: done :+1:

File: traiders/backend/api/models/portfolio.py (hunk 1 above)
Comment (harun-eren) [reply traiders/backend/api/models/portfolio.py]: done :+1:

File: traiders/backend/api/views/portfolio.py (hunk 2 above)
Comment (harun-eren) [reply traiders/backend/api/views/portfolio.py]: done :+1:

File: traiders/backend/api/serializers/portfolio.py (hunk 3 above)
Comment (harun-eren) [reply traiders/backend/api/serializers/portfolio.py]: done :+1:

**Pull request merged**